Iss1020 ecal energy calibration modifications

forcar opened the pull request:
This ECEngine from this iss1020 branch was used to perform pass2 energy and timing calibrations for RG-M. 

https://logbooks.jlab.org/entry/4108190

forcar committed:
ECEngine.java

o setUsePass2Energy to enable usage of pass2 attenuation corrections.
o Update ecTables with /calibration/ec/attenpass2

ECstrip.java

o Abstract class EnergyCorrection
o Energy setters,getters modified
o Two instances pass1Energy, pass2Energy with getEcorr encapsulating
attenuation parameterization

ECCommon.java

o boolean usePass2Energy set to true
o pass1 and pass2 atten tables initialized from ConstantsManager
o setters for pass1,pass2 atten tables

forcar committed:
ECCommon.java

o Add usePass2Energy=false to if(isMC){...} for validation tests

raffaelladevita commented:
I reviewed the changes and, technically, they all look fine. However, I have a couple of suggestions on the naming of the new ccdb table and associated methods:
- I would not put "pass2" in the name of the new table: if it is going to be used for the first time in pass2, the table will remain and be used presumably beyond that;
- same for the methods to handle the corresponding attenuation-length correction;
- in the new attenuation length correction, wouldn't be better to move the 40 and 400 cm to the ccdb table?
- related to the previous, what about creating a table with more parameters so that the same analytical formula could incorporate both the old and new corrections? something like
A*exp(-x/B)+C+D*exp(-x/E)
This would allow to go back to the old correction just loading the proper parameters to this extended table.

forcar committed:
ECEngine.java
o New 5 parameter table atten to replace attenpass2

ECCommon.java
o Implement new table atten, update atten1,atten2 setters

ECstrip.java
o Eliminate pass2 notation for energy calibration method
o Use single attenuation parameterization with table defined
associations (can use either atten1 or atten2 tables depending on
boolean usePass2Energy)

raffaelladevita commented on reconstruction/ec/src/main/java/org/jlab/service/ec/ECCommon.java:
```diff
@@ -150,11 +151,12 @@ public static List<ECStrip>  initEC(DataEvent event,  ConstantsManager manager){
         isMC = run<=100;
         
-        if(isMC) {usePass2Timing = false; useDTCorrections = false; useFTpcal = false;}
+        if(isMC) {usePass2Timing = false; usePass2Energy = false; useDTCorrections = false; useFTpcal = false;}
```
To move away from this iff statement, can the new attenuation table be populated for runs <100 so that the new formula can be used for MC?

forcar replied:
Done.

forcar committed:
Update ECCommon.java

Modified /calibration/ec/atten table for runs 0-100 to mimic the pass1 attenuation table used in GEMC for MC while using the new pass2 formula.   
Removed usePass2Energy=false from line 154.

raffaelladevita committed:
Merge branch 'development' into iss1020-ecal-energy-calibration-modifications
